config.js

```diff
@@ -488,27 +420,45 @@ function testDeprecatedArgv(optimistArgv, komisFamilyNameString) {
         ];
     for (let [oldName, newName] of deprecatedArgv) {
         if (optimistArgv[oldName]) {
-            console.log(`Deprecated: --${oldName} is no longer supported, use --${newName} instead.`);
+            throw new `Deprecated: --${oldName} is no longer supported, use --${newName} instead.`;
         }
     }
     for (let komisGRUArg of familyArrayNamesGRU(komisFamilyNameString)) {
         if (optimistArgv[komisGRUArg]) {
             for (let komi of ["auto","null"]) {
                 if (optimistArgv[komisGRUArg].split(",").includes(komi)) {
-                    console.log(`Deprecated: --${komisGRUArg} /${komi}/ is no longer supported, use --${komisGRUArg} /automatic/ instead`);
+                    throw new `Deprecated: --${komisGRUArg} /${komi}/ is no longer supported, use --${komisGRUArg} /automatic/ instead`;
                 }
             }
         }
     }
     console.log("\n");
 }
 
+// exports arrays:
+function full_r_u_ArgsFromFamilyNameStrings(full_r_u_Families) {
+    let finalArray = [];
+    for (let familyNameString of full_r_u_Families) {
+        familyArrayNamesGRU(familyNameString).forEach(e => finalArray.push(e));
+    }
+    return finalArray;
+}
 
 // optimist.argv.arg(general/ranked/unranked) to exports.(r_u).arg:
 function familyArrayNamesGRU(familyNameString) {
     return ["", "ranked", "unranked"].map(e => `${familyNameString}${e}`);
 }
 
+function argObject_r_u(optimistArgv, familyNameString) {
+    const [general,ranked,unranked] = familyArrayNamesGRU(familyNameString).map(e => optimistArgv[e]);
+    if (general && !ranked && !unranked) {
+        return {ranked: general, unranked: general};
+    } else {
+        return {ranked: (ranked || false),
+                unranked: (unranked || false)};
```

Comment:
What about the case where `ranked === true` and `unranked === undefined`? Or `ranked === undefined` and `unranked === true`? Shouldn't it be `unranked: (unranked || general)`?

Reply:
the problem with `ranked || general` is that this is not the intended behaviour, we wanted : 
- either general arg to be both ranked and unranked
- OR use both ranked and unranked args

if the general arg is smartly selectively replacing ranked or unranked, it would make usage confusing

so in this case roy7 and you said last time bot admin is expected to know how to use ranked and unranked arguments, so this would mean all ranked values are allowed if ranked is undefined i think